### PR TITLE
fix: hogql tests should run through the query cleaning code to avoid flaps

### PR DESCRIPTION
**Pull request by pauldambra**
this flapping test has bit me several times today [`e1220f0` (#25898)](https://github.com/PostHog/posthog/pull/25898/commits/e1220f09e89bea93630f684a3690b8a77db6a743)

and it's confusing to have sql snapshots that don't go through the sql snapshot cleaning code

we aren't using the query cleaning in hogql test sql snapshots, so we periodically flap the snapshots
flapping is slow
so we should avoid it

### CODE REVIEW

File: posthog/hogql/transforms/test/__snapshots__/test_in_cohort.ambr [hunk 1]
```diff
@@ -31,8 +31,8 @@
   FROM events LEFT JOIN (
   SELECT person_static_cohort.person_id AS cohort_person_id, 1 AS matched, person_static_cohort.cohort_id AS cohort_id 
   FROM person_static_cohort 
-  WHERE and(equals(person_static_cohort.team_id, 420), in(person_static_cohort.cohort_id, [6]))) AS __in_cohort ON equals(__in_cohort.cohort_person_id, events.person_id)
```
Comment (pauldambra) [inline posthog/hogql/transforms/test/__snapshots__/test_in_cohort.ambr:34]: this for example was flapping pretty frequently on cohort id on unrelated changes for me

File: posthog/test/base.py [hunk 2]
```diff
@@ -105,6 +105,166 @@
...
+        # replace "uuid" IN ('00000000-0000-4000-8000-000000000001'::uuid) effectively:
+        query = re.sub(
+            r"\"uuid\" IN \('[0-9a-f-]{36}'(::uuid)?(, '[0-9a-f-]{36}'(::uuid)?)*\)",
+            r""""uuid" IN ('00000000-0000-0000-0000-000000000000'::uuid, '00000000-0000-0000-0000-000000000001'::uuid /* ... */)\n""",
+            query,
+        )
+
+    else:
+        query = re.sub(r"(team|cohort)_id(\"?) = \d+", r"\1_id\2 = 2", query)
+        query = re.sub(r"\d+ as (team|cohort)_id(\"?)", r"2 as \1_id\2", query)
```
Comment (Twixes) [inline posthog/test/base.py:123]: The only thing that feels weird here is that this replacement dummy value `2` look just like normal one 😅 whereas `0000000-0000-0000-0000-000000000000` seems obviously like a test placeholder. Something like `99999` would be more obvious (and `2` has been here before this PR, so I'm introducing scope creep)

**Review by Twixes — APPROVED** [review]

File: posthog/test/base.py (hunk 2 above)
Comment (pauldambra) [reply posthog/test/base.py]: only a tiny bit of creep though

your wish is my command

Comment (pauldambra) [reply posthog/test/base.py]: 🤣 111 files changed 📈

**Review by Twixes — APPROVED** [review]
The large number of lines removed and added means this is extremely productive
